### PR TITLE
Added bankers algorithm in java

### PR DESCRIPTION
**Pull request by shraiyya**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Pull request corresponding to issue number #1806 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Please add the DWOC tag.

### CODE REVIEW

File: Java/Operating System/banker.java [hunk 1]
```diff
@@ -0,0 +1,338 @@
+import java.util.Scanner;
```
Comment (HarshCasper) [inline Java/Operating System/banker.java]: Change the Directory name from `Operating System` to `Operating-System`

File: Java/Operating System/banker.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+class algorithm //class for calculation and display
```
Comment (HarshCasper) [inline Java/Operating System/banker.java]: Comments don't have to be very layman. We all understand what work does a class do.

File: Java/Operating System/banker.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+{ 
+ 
+void display(int r,int p,int[]avail,int max[][],int need[][],int allocation[][],int res[])
+{
+	System.out.println("THE MAXIMUM RESOURCES MATRIX IS AS FOLLOWS :");
```
Comment (HarshCasper) [inline Java/Operating System/banker.java]: Follow a consistent indentation

File: Java/Operating System/banker.java [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+    		sum=sum+allocation[j][i];
+    		avail[i]=res[i]-allocation[j][i];
+    	}
+    	avail[i]=res[i]-sum;
+    	System.out.print(avail[i]+"  ");
+    }   
+    System.out.println(" ");
+    System.out.println(" ");
+    
+for (int i = 0 ; i < p ; i++) {
```
Comment (HarshCasper) [inline Java/Operating System/banker.java]: Inconsistent indentation!

File: Java/Operating System/banker.java [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+    	
+        for (int j = 0 ; j < r ; j++) {
+  
+            need[i][j] = max[i][j] - allocation[i][j]; // Need of instance = maxm instance -allocated instance
```
Comment (HarshCasper) [inline Java/Operating System/banker.java]: Use comments above the line of code, not beside them

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by shraiyya** [comment]
I am done with the changes. Please review it again.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Remove the Huffman encoding program, you may have a separate PR for this.

File: Java/README.md [hunk 6]
```diff
@@ -179,6 +179,9 @@ _add list here_
 
 _add list here_
 
+## Operating Systems
+-  [Banker's Algorithm](Operating%20System/banker.java)
```
Comment (iamrajiv) [inline Java/README.md]: @shraiyya This link is not working. Update this to  `-  [Banker's Algorithm](Operating-System/banker.java)`.

File: Java/Operating-System/banker.java [hunk 7]
```diff
@@ -0,0 +1,311 @@
+import java.util.Scanner;
```
Comment (iamrajiv) [inline Java/Operating-System/banker.java:14]: @shraiyya Add description about Banker's algorithm.

File: Java/Operating-System/banker.java [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+        System.out.println("");
+
+
+        System.out.print("AVAILABLE ARRAY IS :");
+        for (int i = 0; i < r; i++) {
+            int sum = 0;
+            avail[i] = 0;
+            for (int j = 0; j < p; j++) {
+                sum = sum + allocation[j][i];
+                avail[i] = res[i] - allocation[j][i];
```
Comment (iamrajiv) [inline Java/Operating-System/banker.java:61]: @shraiyya You can add a comment here about `Checking for available resources inline`.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@shraiyya Add space and time complexity. Also, try to give as many comments about the code.

**Review by atarax665 — APPROVED** [review]

**Review by iamrajiv — APPROVED** [review]

**Comment by shraiyya** [comment]
Can you please label this pull request as DWOC?
Also please add the level.

**Comment by shraiyya** [comment]
Please add the level of the PR for DWOC

**Pull request merged**